src/treq/content.py

```diff
@@ -26,33 +28,74 @@ def _encoding_from_headers(headers):
         return 'UTF-8'
 
 
+class _NullTransport(object):
+    @staticmethod
+    def pauseProducing():
+        pass
+
+    @staticmethod
+    def resumeProducing():
+        pass
+
+    @staticmethod
+    def stopProducing():
+        pass
+
+
 class _BodyCollector(Protocol):
     def __init__(self, finished, collector):
+        self.buffer = b''
```

Comment:
There's no need to have a buffer unless you expect that `dataRecieved` will be called incorrectly by a suspended protocol?  (if that happens it's probably time to shut down the connection, since there are ordering questions that bug raises)